lib/logstash/outputs/elasticsearch_java.rb

```diff
@@ -328,122 +305,70 @@ def register
     @retry_queue_not_full = ConditionVariable.new
     @retry_queue = Queue.new
 
-    client_settings = {}
-
-
-    if @protocol.nil?
-      @protocol = LogStash::Environment.jruby? ? "node" : "http"
-    end
-
-    if @protocol == "http"
-      if @action == "create_unless_exists"
-        raise(LogStash::ConfigurationError, "action => 'create_unless_exists' is not supported under the HTTP protocol");
-      end
 
-      client_settings[:path] = "/#{@path}/".gsub(/\/+/, "/") # Normalize slashes
-      @logger.debug? && @logger.debug("Normalizing http path", :path => @path, :normalized => client_settings[:path])
+    if @protocol ==='node' && !@network_host
+      raise LogStash::ConfigurationError, "network_host MUST be set if the 'node' protocol is in use!"
```

Comment:
Maybe make this more clear that it's required because otherwise Elasticsearch won't know what address to advertise for itself or what address(es) to bind to.